// code-auth/index.js
var twilio = require('twilio');
var nodemailer = require('nodemailer');

module.exports.sendAuthCode = function(authCode, medium){
		switch(medium){
			case 'sms':
				sendSms(authCode)
				break;

			case 'email':
				sendEmail(authCode)
				break;

			default:
				console.log(authCode)			
		}
	}

var sendEmail = function(authCode){
	// create reusable transporter object using the default SMTP transport
	var transporter = nodemailer.createTransport({
	    service: 'gmail',
	    auth: {
	        user: 'tolmanbldgpsychology@gmail.com',
	        pass: 'T0lman94720'
	    }
	});;

		// setup email data with unicode symbols
	var mailOptions = {
	    from: 'tolmanbldgpsychology@gmail.com', // sender address
	    to: 'mfaraji@berkeley.edu', // list of receivers
	    subject: 'authCode', // Subject line
	    text: 'authCode: ' + authCode, // plain text body
	    //html: '<b>Hello world ?</b>' // html body
	};

	// send mail with defined transport object
	transporter.sendMail(mailOptions, (error, info) => {
    if (err) throw err;   
	    //console.log('Message %s sent: %s', info.messageId, info.response);
	});	
}

var sendSms = function(authCode){

	// Find your account sid and auth token in your Twilio account Console.
	var client = twilio('AC1d17545485a49ebd98279b906a7c3e60', '22ad445f62344171f01768937fcbfd10');
	 
	// Send the text message.
	client.sendMessage({
	  to: '+18103940029',
	  from: '+15005550006',
	  body: 'authCode: ' + authCode
	}, function(err, res){
	    //console.log('Messages.send()', err, res);
			if (err) throw err; 				
	});
}
